exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestNewSimpleRepeatedFunctions.java

```diff
@@ -17,36 +17,161 @@
+      {"mapArray": [{"field1": 1}, {"field1": 2}]}
+      {"mapArray": [{"field2": "val2"}, {"field1": 2}, {"field1": 2}]}
+      {"mapArray": []}
+      {"mapArray": [{"field1": 1, "field2": "val3"}]}
+      {"mapArray": [{"field1": 1, "field2": "val4"}, {"field1": 2}, {"field1": 2}, {"field2": "val1"}, {"field1": 2}]}
+      {"mapArray": [{"field1": 1, "field2": "val3"}, {"field1": 2}]}
+    */
+    String fileName = "repeated_count_map.json";
+    try (BufferedWriter writer = new BufferedWriter(new FileWriter(
+        new File(dirTestWatcher.getRootDir(), fileName)))) {
+      String[] arrayElements = {
+          "{\"field1\": 1, \"field2\": \"val1\"}, {\"field1\": 2}",
+          "{\"field1\": 1}, {\"field1\": 2}",
+          "{\"field2\": \"val2\"}, {\"field1\": 2}, {\"field1\": 2}",
+          "",
+          "{\"field1\": 1, \"field2\": \"val3\"}",
+          "{\"field1\": 1, \"field2\": \"val4\"}, {\"field1\": 2}, {\"field1\": 2}, {\"field2\": \"val1\"}, {\"field1\": 2}",
+          "{\"field1\": 1, \"field2\": \"val3\"}, {\"field1\": 2}"
+      };
+      for (String value : arrayElements) {
+        String entry = String.format("{\"mapArray\": [%s]}\n", value);
+        writer.write(entry);
+      }
+    }
+
+    String selectQuery = "select repeated_count(mapArray) from dfs.`%s`";
+    testBuilder()
+        .sqlQuery(selectQuery, fileName)
+        .ordered()
+        .baselineColumns("EXPR$0")
+        .baselineValuesForSingleColumn(2, 2, 3, 0, 1, 5, 2)
+        .go();
+
+    testBuilder()
+        .sqlQuery(selectQuery + " where repeated_count(mapArray) > 2", fileName)
+        .ordered()
+        .baselineColumns("EXPR$0")
+        .baselineValuesForSingleColumn(3, 5)
+        .go();
+
+    testBuilder()
+        .sqlQuery(selectQuery + " group by 1", fileName)
+        .unOrdered()
+        .baselineColumns("EXPR$0")
+        .baselineValuesForSingleColumn(2, 3, 0, 1, 5)
+        .go();
+
+    testBuilder()
+        .sqlQuery(selectQuery + " group by 1 having repeated_count(mapArray) < 3", fileName)
+        .unOrdered()
+        .baselineColumns("EXPR$0")
+        .baselineValuesForSingleColumn(2, 0, 1)
+        .go();
+  }
+
+  @Test
+  public void testRepeatedCountRepeatedList() throws Exception {
+    // Contents of the generated file:
+    /*
+      {"id": 1, "array": [[1, 2], [1, 3], [2, 3]]}
+      {"id": 2, "array": []}
+      {"id": 3, "array": [[2, 3], [1, 3, 4]]}
+      {"id": 4, "array": [[1], [2], [3, 4], [5], [6]]}
+      {"id": 5, "array": [[1, 2, 3], [4, 5], [6], [7], [8, 9], [2, 3], [2, 3], [2, 3], [2]]}
+      {"id": 6, "array": [[1, 2], [3], [4], [5]]}
+      {"id": 7, "array": []}
+      {"id": 8, "array": [[1], [2], [3]]}
+    */
+    String fileName = "repeated_count_list.json";
+    try (BufferedWriter writer = new BufferedWriter(new FileWriter(
+        new File(dirTestWatcher.getRootDir(), fileName)))) {
+      String[] arrayElements = {
+          "[1, 2], [1, 3], [2, 3]",
```

Comment:
Use `TestBuilder.listOf()` to create a `JsonStringArrayList` from the given values.